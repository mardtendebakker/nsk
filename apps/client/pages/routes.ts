export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
