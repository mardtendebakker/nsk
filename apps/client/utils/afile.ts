import { AFile } from './axios/models/aFile';

export function buildAFileLink(afile: AFile): string {
  return `https://${afile.unique_server_filename}.s3.amazonaws.com/${afile.discr}/${afile.original_client_filename}`;
}
