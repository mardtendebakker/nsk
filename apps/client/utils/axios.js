import axios from 'axios';

export { AxiosPromise, AxiosResponse, AxiosError } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333' : 'production_url';

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const SUPLIERS_PATH = '';

export default client;
