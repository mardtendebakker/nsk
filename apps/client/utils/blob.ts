export function openBlob(blob: Blob): void {
  const a = document.createElement('a');
  document.body.appendChild(a);
  a.style.cssText = 'display: none';

  const url = window.URL.createObjectURL(blob);

  a.href = url;
  a.target = '_blank';
  a.click();
  window.URL.revokeObjectURL(url);
}
