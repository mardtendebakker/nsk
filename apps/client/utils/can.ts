import _ from 'lodash';
import { Group } from '../stores/security/types';

export default function can(userGroups: Group[], requiredGroups: Group[], disableDefaultGroups = false): boolean {
  return _.intersection(userGroups, disableDefaultGroups ? requiredGroups : [...requiredGroups, 'admin', 'super_admin']).length > 0;
}
