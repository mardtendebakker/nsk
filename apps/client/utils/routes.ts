import { Group, User } from '../stores/security/types';

export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
export const BULK_EMAIL = '/bulk-email';
export const BULK_EMAIL_NEW = '/bulk-email/new';
export const CONTACTS_EDIT = '/contacts/[id]';
export const CONTACTS_NEW = '/contacts/new';
export const CONTACTS = '/contacts';
export const COMPANIES_EDIT = '/companies/[id]';
export const COMPANIES_CONTACTS_EDIT = '/companies/[id]/contacts/[contact_id]';
export const COMPANIES_CONTACTS_NEW = '/companies/[id]/contacts/new';
export const COMPANIES_NEW = '/companies/new';
export const COMPANIES = '/companies';
export const ACCOUNT_VERIFICATION = '/account-verification';
export const SETTINGS = '/settings';
export const ORDERS_PURCHASES = '/orders/purchases';
export const ORDERS_SALES = '/orders/sales';
export const ORDERS_REPAIRS = '/orders/repairs';
export const STOCKS_PRODUCTS = '/stock/products';
export const ORDERS_PURCHASES_NEW = '/orders/purchases/new';
export const ORDERS_PURCHASES_EDIT = '/orders/purchases/[id]';
export const ORDERS_REPAIRS_EDIT = '/orders/repairs/[id]';
export const ORDERS_SALES_NEW = '/orders/sales/new';
export const ORDERS_REPAIRS_NEW = '/orders/repairs/new';
export const ORDERS_SALES_EDIT = '/orders/sales/[id]';
export const STOCKS_REPAIR_SERVICES = '/stock/repair-services';
export const STOCKS_ARCHIVED = '/stock/archived';
export const MY_TASKS = '/my-tasks';
export const ADMIN_USERS = '/admin/users';
export const ADMIN_SETTINGS = '/admin/settings';
export const ADMIN_SETTINGS_LOCATIONS = '/admin/settings/locations';
export const ADMIN_SETTINGS_PRODUCT_TYPES = '/admin/settings/product-types';
export const ADMIN_SETTINGS_ATTRIBUTES = '/admin/settings/attributes';
export const ADMIN_SETTINGS_TASKS = '/admin/settings/tasks';
export const ADMIN_SETTINGS_STATUS = '/admin/settings/status';
export const ADMIN_SETTINGS_MAILING_LISTS = '/admin/settings/mailing-lists';
export const ADMIN_SETTINGS_CUSTOMER_TAGS = '/admin/settings/customer-tags';
export const ADMIN_SETTINGS_PRODUCT_STATUSES = '/admin/settings/product-statuses';
export const ADMIN_SETTINGS_ORDER_STATUSES = '/admin/settings/order-statuses';
export const LOGISTICS_PICKUP = '/logistics/pickups';
export const LOGISTICS_DELIVERY = '/logistics/deliveries';

export const getRouteGroups = (uri: string): Group[] => {
  if (uri.startsWith('/admin')) {
    return ['admin', 'super_admin'];
  }

  if (uri.startsWith('/bulk-email')) {
    return ['admin', 'super_admin', 'manager'];
  }

  if (uri.startsWith('/logistics')) {
    return ['admin', 'super_admin', 'manager', 'logistics'];
  }

  if (
    uri.startsWith('/stock')
    || uri.startsWith('/my-tasks')
    || uri.startsWith('/bulk-email')
  ) {
    return ['admin', 'super_admin', 'manager', 'logistics', 'local'];
  }

  if (
    uri.startsWith('/orders')
    || uri.startsWith('/contacts')
  ) {
    return ['admin', 'super_admin', 'manager', 'logistics', 'local', 'partner_sale_uploader', 'partner'];
  }
};

export const getDefaultPath = (user: User) => DASHBOARD;
