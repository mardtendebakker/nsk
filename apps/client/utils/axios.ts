import axios from 'axios';

export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333/api' : process.env.NX_AXIOS_BASE_URL;

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const CUSTOMERS_PATH = '/customers/:id';
export const SUPPLIERS_PATH = '/suppliers/:id';

export default client;

export interface Customer {
  id: number,
  name?: string,
  representative?: string,
  kvk_nr?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  is_partner?: number,
}

export interface Supplier {
  id: number,
  name?: string,
  representative?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  partner?: string,
}
