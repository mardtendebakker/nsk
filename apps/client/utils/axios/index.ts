import axios from 'axios';
import axiosClient from './client';
import refreshToken from './refreshTokenInterceptor';

export {
  CUSTOMERS_PATH,
  SUPPLIERS_PATH,
  SIGN_IN_PATH,
  REFRESH_TOKEN_PATH,
  USER_INFO_PATH,
  SIGN_UP_PATH,
  CONFIRM_ACCOUNT_PATH,
  CHANGE_PASSWORD_PATH,
  SEND_VERIFICATION_CODE_PATH,
  FORGOT_PASSWORD_PATH,
  SALES_ORDERS_PATH,
  PURCHASE_ORDERS_PATH,
  STOCK_PRODUCTS_PATH,
  USERS_PATH,
  STOCK_REPAIR_SERVICES_PATH,
  ORDER_STATUSES_PATH,
  COMPANIES_PATH,
  LOCATIONS_PATH,
  BULK_PRINT_SALES_PATH,
  BULK_PRINT_PURCHASES_PATH,
  ORDERS_ANALYTICS_PATH,
} from './paths';
export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

axiosClient.interceptors.response.use((response) => response, refreshToken);

export const { CancelToken } = axios;

export default axiosClient;
