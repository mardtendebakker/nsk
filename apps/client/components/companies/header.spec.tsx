import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Header from './header';
import { ADMIN_USERS } from '../../utils/routes';

const mockRouter = {
  push: jest.fn(),
  get pathname() { return ADMIN_USERS; },
};

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

const mockUseSecurity = {
  hasModule: jest.fn(() => true),
};

jest.mock('../../hooks/useSecurity', () => jest.fn(() => mockUseSecurity));

describe('Header component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders properly', () => {
    const { getByText } = render(<Header />);

    expect(getByText('companies')).toBeInTheDocument();
    expect(getByText('newCompany')).toBeInTheDocument();
  });
  it('pushed to new company page properly', () => {
    const { getByText } = render(<Header />);
    fireEvent.click(getByText('newCompany'));
    expect(mockRouter.push).toHaveBeenCalledWith('/companies/new');
  });
  it('don\'t push to new company page if module not active', () => {
    mockUseSecurity.hasModule.mockReturnValue(false);

    const { getByText } = render(<Header />);
    fireEvent.click(getByText('newCompany'));
    expect(mockRouter.push).not.toBeCalled();
    expect(mockUseSecurity.hasModule).toBeCalledWith('customer_contact_action');
  });
});
