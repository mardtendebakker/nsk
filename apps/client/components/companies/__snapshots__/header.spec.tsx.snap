// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1su3en"
  >
    <div
      class="MuiBox-root css-1dtnjt5"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-xou334-MuiTypography-root"
      >
        companies
      </h4>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-5tl3og-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      newCompany
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
